src/service/fightData.ts

```diff
@@ -0,0 +1,223 @@
+        lossDescription: "",
+        winDescription: "",
+        health: 350,
+        enabled: false,
+        baseDamage: 5,
+        baseDefence: 5,
+        items: [],
+    },
+};
+
+export const baseStats = {
+    description: "",
+    health: 80,
+    baseDamage: 1,
+    baseDefence: 0,
+};
+
+export type FightItemType = "weapon" | "armor" | "item";
+
+export type Equipable = EquipableWeapon | EquipableItem | EquipableArmor;
+
+export interface EquipableWeapon {
+    type: "weapon";
+    attack: Range;
+}
+
+export interface EquipableArmor {
+    type: "armor";
+    defence: Range;
+    health: number;
+}
+
+export interface FightScene {
+    player: Entity;
+    enemy: Entity;
+}
+
+export interface BaseEntity {
+    health: number;
+    name: string;
+    description: string;
+    baseDamage: number;
+    baseDefence: number;
+
+    items: (EquipableItem & { name: string })[];
+    weapon?: EquipableWeapon & { name: string };
+    armor?: EquipableArmor & { name: string };
+    //TODO
+    permaBuffs?: undefined;
+}
+
+export interface Enemy extends BaseEntity {
+    enabled: boolean;
+    winDescription: string;
+    lossDescription: string;
+}
+
+export class Entity {
+    stats: BaseEntity;
+    maxhealth: number;
+    lastattack?: number;
+    lastdefence?: number;
+    itemtext: string[] = [];
+
+    constructor(entity: BaseEntity) {
+        this.stats = entity;
+        if (this.stats.armor?.health) {
+            this.stats.health += this.stats.armor?.health;
+        }
+        this.maxhealth = this.stats.health;
+    }
+
+    attack(enemy: Entity) {
+        let rawDamage: number;
+        rawDamage = this.stats.baseDamage;
+        if (this.stats.weapon?.attack) {
+            rawDamage += randomValue(this.stats.weapon.attack);
+        }
+        for (const value1 of this.stats.items) {
+            if (value1.attackModifier) {
+                rawDamage += randomValue(value1.attackModifier);
+            }
+        }
+        const defence = enemy.defend();
+        const result = calcDamage(rawDamage, defence);
+        console.log(
+            `${this.stats.name} (${this.stats.health}) hits ${enemy.stats.name} (${enemy.stats.health}) for ${result.damage} mitigated ${result.mitigated}`,
+        );
+        enemy.stats.health -= result.damage;
+        this.lastattack = result.rawDamage;
+        return result;
+    }
+
+    defend() {
+        let defence = this.stats.baseDefence;
+        if (this.stats.armor?.defence) {
+            defence += randomValue(this.stats.armor.defence);
+        }
+        for (const item of this.stats.items) {
+            if (item.defenceModifier) {
+                defence += randomValue(item.defenceModifier);
+            }
+        }
+        this.lastdefence = defence;
+        return defence;
+    }
+}
+
+export interface Range {
+    min: number;
+    max: number;
+}
```

Comment:
Das + `randomValue` is cool, das sollten wir in ne extra datei auslagern, `src/service/random.ts` oder so. Das können wir auch an anderen Stellen noch benutzen